log4j-core/src/main/java/org/apache/logging/log4j/core/config/ConfigurationFactory.java

```diff
@@ -294,7 +302,22 @@ static String extractClassLoaderUriPath(final URI uri) {
     protected ConfigurationSource getInputFromString(final String config, final ClassLoader loader) {
         try {
             final URL url = new URL(config);
-            return new ConfigurationSource(url.openStream(), FileUtils.fileFromUri(url.toURI()));
+            URLConnection urlConnection = url.openConnection();
+            if (url.getProtocol().equals(HTTPS)) {
+                SslConfiguration sslConfiguration = SslConfigurationFactory.getSslConfiguration();
+                if (sslConfiguration != null) {
+                    ((HttpsURLConnection) urlConnection).setSSLSocketFactory(sslConfiguration.getSslSocketFactory());
+                    if (!sslConfiguration.isVerifyHostName()) {
+                        ((HttpsURLConnection) urlConnection).setHostnameVerifier(LaxHostnameVerifier.INSTANCE);
```

Comment:
I mean, at least you're not setting it globally, but why is this here?

Reply:
HostNameVerifier? I copied this code from the HttpAppender but since we haven't configured yet I had to get creative in how the SSL settings would be obtained.

Reply:
It is handled in SslConfigurationFactory.